### PR TITLE
bugfix: sql statement escape symbol

### LINKED ISSUE #1665: table use keyword as primary key field cause InsertExecutor afterImage error
- [x] I have searched the [issues](https://github.com/seata/seata/issues) of this repository and believe that this is not a duplicate.

### Ⅰ. Issue Description
```
create table test_keyword(
  `in` varchar(45) primary key,
  `desc` varchar(45)
)

insert into test_keyword values(?, ?)
```

### Ⅱ. Describe what happened

  If there is an exception, please attach the exception trace:

```
java.sql.SQLSyntaxErrorException: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'in='aa65ac42-4fcc-4412-a256-108841dd21f1'' at line 1
at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:536) ~[mysql-connector-java-6.0.6.jar:6.0.6]
	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:513) ~[mysql-connector-java-6.0.6.jar:6.0.6]
	at com.mysql.cj.jdbc.exceptions.SQLExceptionsMapping.translateException(SQLExceptionsMapping.java:115) ~[mysql-connector-java-6.0.6.jar:6.0.6]
	at com.mysql.cj.jdbc.ConnectionImpl.execSQL(ConnectionImpl.java:1983) ~[mysql-connector-java-6.0.6.jar:6.0.6]
	at com.mysql.cj.jdbc.PreparedStatement.executeInternal(PreparedStatement.java:1826) ~[mysql-connector-java-6.0.6.jar:6.0.6]
	at com.mysql.cj.jdbc.PreparedStatement.executeQuery(PreparedStatement.java:1923) ~[mysql-connector-java-6.0.6.jar:6.0.6]
	at com.alibaba.druid.pool.DruidPooledPreparedStatement.executeQuery(DruidPooledPreparedStatement.java:228) ~[druid-1.1.10.jar:1.1.10]
	at io.seata.rm.datasource.exec.BaseTransactionalExecutor.buildTableRecords(BaseTransactionalExecutor.java:340) ~[seata-all-0.8.1-SNAPSHOT.jar:0.8.1-SNAPSHOT]
	at io.seata.rm.datasource.exec.InsertExecutor.afterImage(InsertExecutor.java:80) ~[seata-all-0.8.1-SNAPSHOT.jar:0.8.1-SNAPSHOT]
	at io.seata.rm.datasource.exec.AbstractDMLBaseExecutor.executeAutoCommitFalse(AbstractDMLBaseExecutor.java:74) ~[seata-all-0.8.1-SNAPSHOT.jar:0.8.1-SNAPSHOT]
	at io.seata.rm.datasource.exec.AbstractDMLBaseExecutor.lambda$executeAutoCommitTrue$0(AbstractDMLBaseExecutor.java:91) ~[seata-all-0.8.1-SNAPSHOT.jar:0.8.1-SNAPSHOT]
	at io.seata.rm.datasource.ConnectionProxy$LockRetryPolicy.doRetryOnLockConflict(ConnectionProxy.java:290) ~[seata-all-0.8.1-SNAPSHOT.jar:0.8.1-SNAPSHOT]
	at io.seata.rm.datasource.exec.AbstractDMLBaseExecutor$LockRetryPolicy.execute(AbstractDMLBaseExecutor.java:133) ~[seata-all-0.8.1-SNAPSHOT.jar:0.8.1-SNAPSHOT]
	at io.seata.rm.datasource.exec.AbstractDMLBaseExecutor.executeAutoCommitTrue(AbstractDMLBaseExecutor.java:90) ~[seata-all-0.8.1-SNAPSHOT.jar:0.8.1-SNAPSHOT]
	at io.seata.rm.datasource.exec.AbstractDMLBaseExecutor.doExecute(AbstractDMLBaseExecutor.java:58) ~[seata-all-0.8.1-SNAPSHOT.jar:0.8.1-SNAPSHOT]
	at io.seata.rm.datasource.exec.BaseTransactionalExecutor.execute(BaseTransactionalExecutor.java:96) ~[seata-all-0.8.1-SNAPSHOT.jar:0.8.1-SNAPSHOT]
	at io.seata.rm.datasource.exec.ExecuteTemplate.execute(ExecuteTemplate.java:101) ~[seata-all-0.8.1-SNAPSHOT.jar:0.8.1-SNAPSHOT]
	at io.seata.rm.datasource.exec.ExecuteTemplate.execute(ExecuteTemplate.java:47) ~[seata-all-0.8.1-SNAPSHOT.jar:0.8.1-SNAPSHOT]
	at io.seata.rm.datasource.PreparedStatementProxy.executeUpdate(PreparedStatementProxy.java:74) ~[seata-all-0.8.1-SNAPSHOT.jar:0.8.1-SNAPSHOT]
	at org.springframework.jdbc.core.JdbcTemplate.lambda$update$0(JdbcTemplate.java:866) ~[spring-jdbc-5.1.9.RELEASE.jar:5.1.9.RELEASE]
	at org.springframework.jdbc.core.JdbcTemplate.execute(JdbcTemplate.java:616) ~[spring-jdbc-5.1.9.RELEASE.jar:5.1.9.RELEASE]
	at org.springframework.jdbc.core.JdbcTemplate.update(JdbcTemplate.java:861) ~[spring-jdbc-5.1.9.RELEASE.jar:5.1.9.RELEASE]
	at org.springframework.jdbc.core.JdbcTemplate.update(JdbcTemplate.java:916) ~[spring-jdbc-5.1.9.RELEASE.jar:5.1.9.RELEASE]
	at org.springframework.jdbc.core.JdbcTemplate.update(JdbcTemplate.java:926) ~[spring-jdbc-5.1.9.RELEASE.jar:5.1.9.RELEASE]
	at org.xxz.test.service.TestService.test8(TestService.java:97) ~[classes/:na]
	at org.xxz.test.service.TestService$$FastClassBySpringCGLIB$$f101794c.invoke(<generated>) ~[classes/:na]
	at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218) ~[spring-core-5.1.9.RELEASE.jar:5.1.9.RELEASE]
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:749) ~[spring-aop-5.1.9.RELEASE.jar:5.1.9.RELEASE]
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163) ~[spring-aop-5.1.9.RELEASE.jar:5.1.9.RELEASE]
	at io.seata.spring.annotation.GlobalTransactionalInterceptor$1.execute(GlobalTransactionalInterceptor.java:104) ~[seata-all-0.8.1-SNAPSHOT.jar:0.8.1-SNAPSHOT]
	at io.seata.tm.api.TransactionalTemplate.execute(TransactionalTemplate.java:64) ~[seata-all-0.8.1-SNAPSHOT.jar:0.8.1-SNAPSHOT]
	at io.seata.spring.annotation.GlobalTransactionalInterceptor.handleGlobalTransaction(GlobalTransactionalInterceptor.java:101) ~[seata-all-0.8.1-SNAPSHOT.jar:0.8.1-SNAPSHOT]
	at io.seata.spring.annotation.GlobalTransactionalInterceptor.invoke(GlobalTransactionalInterceptor.java:76) ~[seata-all-0.8.1-SNAPSHOT.jar:0.8.1-SNAPSHOT]
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186) ~[spring-aop-5.1.9.RELEASE.jar:5.1.9.RELEASE]
	at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:688) ~[spring-aop-5.1.9.RELEASE.jar:5.1.9.RELEASE]
	at org.xxz.test.service.TestService$$EnhancerBySpringCGLIB$$6e14b166.test8(<generated>) ~[classes/:na]
	at org.xxz.test.controller.TestController.test8(TestController.java:63) ~[classes/:na]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.8.0_211]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[na:1.8.0_211]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.8.0_211]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[na:1.8.0_211]
	at org.springframework.web.method.support.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:190) ~[spring-web-5.1.9.RELEASE.jar:5.1.9.RELEASE]
	at org.springframework.web.method.support.InvocableHandlerMethod.invokeForRequest(InvocableHandlerMethod.java:138) ~[spring-web-5.1.9.RELEASE.jar:5.1.9.RELEASE]
	at org.springframework.web.servlet.mvc.method.annotation.ServletInvocableHandlerMethod.invokeAndHandle(ServletInvocableHandlerMethod.java:104) ~[spring-webmvc-5.1.9.RELEASE.jar:5.1.9.RELEASE]
	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.invokeHandlerMethod(RequestMappingHandlerAdapter.java:892) ~[spring-webmvc-5.1.9.RELEASE.jar:5.1.9.RELEASE]
	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.handleInternal(RequestMappingHandlerAdapter.java:797) ~[spring-webmvc-5.1.9.RELEASE.jar:5.1.9.RELEASE]
	at org.springframework.web.servlet.mvc.method.AbstractHandlerMethodAdapter.handle(AbstractHandlerMethodAdapter.java:87) ~[spring-webmvc-5.1.9.RELEASE.jar:5.1.9.RELEASE]
	at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:1039) ~[spring-webmvc-5.1.9.RELEASE.jar:5.1.9.RELEASE]
	at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:942) ~[spring-webmvc-5.1.9.RELEASE.jar:5.1.9.RELEASE]
	at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:1005) ~[spring-webmvc-5.1.9.RELEASE.jar:5.1.9.RELEASE]
	at org.springframework.web.servlet.FrameworkServlet.doGet(FrameworkServlet.java:897) ~[spring-webmvc-5.1.9.RELEASE.jar:5.1.9.RELEASE]
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:634) ~[tomcat-embed-core-9.0.22.jar:9.0.22]
	at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:882) ~[spring-webmvc-5.1.9.RELEASE.jar:5.1.9.RELEASE]
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:741) ~[tomcat-embed-core-9.0.22.jar:9.0.22]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:231) ~[tomcat-embed-core-9.0.22.jar:9.0.22]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[tomcat-embed-core-9.0.22.jar:9.0.22]
	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:53) ~[tomcat-embed-websocket-9.0.22.jar:9.0.22]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) ~[tomcat-embed-core-9.0.22.jar:9.0.22]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[tomcat-embed-core-9.0.22.jar:9.0.22]
	at org.springframework.web.filter.RequestContextFilter.doFilterInternal(RequestContextFilter.java:99) ~[spring-web-5.1.9.RELEASE.jar:5.1.9.RELEASE]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:118) ~[spring-web-5.1.9.RELEASE.jar:5.1.9.RELEASE]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) ~[tomcat-embed-core-9.0.22.jar:9.0.22]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[tomcat-embed-core-9.0.22.jar:9.0.22]
	at org.springframework.web.filter.FormContentFilter.doFilterInternal(FormContentFilter.java:92) ~[spring-web-5.1.9.RELEASE.jar:5.1.9.RELEASE]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:118) ~[spring-web-5.1.9.RELEASE.jar:5.1.9.RELEASE]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) ~[tomcat-embed-core-9.0.22.jar:9.0.22]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[tomcat-embed-core-9.0.22.jar:9.0.22]
	at org.springframework.web.filter.HiddenHttpMethodFilter.doFilterInternal(HiddenHttpMethodFilter.java:93) ~[spring-web-5.1.9.RELEASE.jar:5.1.9.RELEASE]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:118) ~[spring-web-5.1.9.RELEASE.jar:5.1.9.RELEASE]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) ~[tomcat-embed-core-9.0.22.jar:9.0.22]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[tomcat-embed-core-9.0.22.jar:9.0.22]
	at org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:200) ~[spring-web-5.1.9.RELEASE.jar:5.1.9.RELEASE]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:118) ~[spring-web-5.1.9.RELEASE.jar:5.1.9.RELEASE]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) ~[tomcat-embed-core-9.0.22.jar:9.0.22]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[tomcat-embed-core-9.0.22.jar:9.0.22]
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:202) ~[tomcat-embed-core-9.0.22.jar:9.0.22]
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:96) [tomcat-embed-core-9.0.22.jar:9.0.22]
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:490) [tomcat-embed-core-9.0.22.jar:9.0.22]
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:139) [tomcat-embed-core-9.0.22.jar:9.0.22]
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:92) [tomcat-embed-core-9.0.22.jar:9.0.22]
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74) [tomcat-embed-core-9.0.22.jar:9.0.22]
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:343) [tomcat-embed-core-9.0.22.jar:9.0.22]
	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:408) [tomcat-embed-core-9.0.22.jar:9.0.22]
	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:66) [tomcat-embed-core-9.0.22.jar:9.0.22]
	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:853) [tomcat-embed-core-9.0.22.jar:9.0.22]
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1587) [tomcat-embed-core-9.0.22.jar:9.0.22]
	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49) [tomcat-embed-core-9.0.22.jar:9.0.22]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [na:1.8.0_211]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [na:1.8.0_211]
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61) [tomcat-embed-core-9.0.22.jar:9.0.22]
	at java.lang.Thread.run(Thread.java:748) [na:1.8.0_211]
```

### Ⅳ. How to reproduce it (as minimally and precisely as possible)

1. xxx
2. xxx
3. xxx

### Ⅵ. Environment:

- JDK version :
- OS :
- Others:

### PR DESCRIPTION
**Pull request by jsbxyyx**
### Ⅰ. Describe what this PR did
bugfix:
1. insert column has escape symbol 
2. insert afterImage primary key is keyword.

### Ⅱ. Does this pull request fix one issue?

fixes #1665 #1667 #1655

### CODE REVIEW

File: rm-datasource/src/main/java/io/seata/rm/datasource/ColumnUtils.java [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+     * @param escape the escape
+     * @throws NullPointerException if cols is null
+     */
+    public static void delEscape(List<String> cols, Escape escape) {
+        if (cols == null) {
+            throw new NullPointerException("cols is null");
+        }
+        for (int i = 0, len = cols.size(); i < len; i++) {
+            String col = cols.get(i);
+            if (col != null && col.charAt(0) == escape.value) {
```
Comment (slievrly) [inline rm-datasource/src/main/java/io/seata/rm/datasource/ColumnUtils.java]: col.chartAt(col.length-1) ?

Comment (jsbxyyx) [reply rm-datasource/src/main/java/io/seata/rm/datasource/ColumnUtils.java]: fix done

File: rm-datasource/src/main/java/io/seata/rm/datasource/AbstractConnectionProxy.java [hunk 2]
```diff
@@ -99,7 +103,15 @@ public Statement createStatement() throws SQLException {
 
     @Override
     public PreparedStatement prepareStatement(String sql) throws SQLException {
-        PreparedStatement targetPreparedStatement = getTargetConnection().prepareStatement(sql, Statement.RETURN_GENERATED_KEYS);
+        String dbType = getDbType();
+        PreparedStatement targetPreparedStatement;
+        // exclude oracle
+        boolean dbTypeEquals = StringUtils.equalsIgnoreCase(JdbcConstants.ORACLE, dbType);
+        if (dbTypeEquals) {
+            targetPreparedStatement = getTargetConnection().prepareStatement(sql);
```
Comment (slievrly) [inline rm-datasource/src/main/java/io/seata/rm/datasource/AbstractConnectionProxy.java]: split to another pr to solve oracle's rowid

File: rm-datasource/src/main/java/io/seata/rm/datasource/sql/struct/TableMetaAdapter.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        }
+
+        List<String> pk = getPrimaryKeyOnlyName();
+        if (pk.isEmpty()) {
+            return false;
+        }
+
+        ColumnUtils.addEscape(cols, dbType);
+
+        if (cols.containsAll(pk)) {
```
Comment (slievrly) [inline rm-datasource/src/main/java/io/seata/rm/datasource/sql/struct/TableMetaAdapter.java]: pk contains escape?

File: rm-datasource/src/main/java/io/seata/rm/datasource/sql/struct/TableMetaCacheAdapter.java [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+        TableMetaAdapter adapter = new TableMetaAdapter();
+
+        adapter.setDbType(dbType);
+        adapter.setTableName(ColumnUtils.addEscape(tableMeta.getTableName(), dbType));
+
+        for (Map.Entry<String, ColumnMeta> entry : tableMeta.getAllColumns().entrySet()) {
+            ColumnMeta value = entry.getValue();
+            if (value.getColumnName() != null) {
+                value.setColumnName(ColumnUtils.addEscape(value.getColumnName(), dbType));
```
Comment (slievrly) [inline rm-datasource/src/main/java/io/seata/rm/datasource/sql/struct/TableMetaCacheAdapter.java]: Why did you modify the original value?

File: rm-datasource/src/main/java/io/seata/rm/datasource/sql/struct/TableMetaAdapter.java (hunk 3 above)
Comment (jsbxyyx) [reply rm-datasource/src/main/java/io/seata/rm/datasource/sql/struct/TableMetaAdapter.java]: yes

File: rm-datasource/src/main/java/io/seata/rm/datasource/sql/struct/TableMetaCacheAdapter.java (hunk 4 above)
Comment (jsbxyyx) [reply rm-datasource/src/main/java/io/seata/rm/datasource/sql/struct/TableMetaCacheAdapter.java]: done

File: rm-datasource/src/main/java/io/seata/rm/datasource/sql/struct/TableMetaAdapter.java (hunk 3 above)
Comment (jsbxyyx) [reply rm-datasource/src/main/java/io/seata/rm/datasource/sql/struct/TableMetaAdapter.java]: done

File: rm-datasource/src/main/java/io/seata/rm/datasource/sql/struct/TableMetaCacheAdapter.java [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+            tableMeta = TableMetaCacheOracle.getTableMeta(dataSourceProxy, tableName);
+        } else {
+            tableMeta = TableMetaCache.getTableMeta(dataSourceProxy, tableName);
+        }
+
+        if (!escape) {
+            return tableMeta;
+        }
+
+        TableMetaAdapter adapter = new TableMetaAdapter();
```
Comment (slievrly) [inline rm-datasource/src/main/java/io/seata/rm/datasource/sql/struct/TableMetaCacheAdapter.java]: TableMetaWrapper？ ...decorate

Comment (jsbxyyx) [reply rm-datasource/src/main/java/io/seata/rm/datasource/sql/struct/TableMetaCacheAdapter.java]: same idea

File: rm-datasource/src/main/java/io/seata/rm/datasource/AbstractConnectionProxy.java [hunk 6]
```diff
@@ -112,14 +113,7 @@ public PreparedStatement prepareStatement(String sql) throws SQLException {
...
         if (sqlRecognizer != null && sqlRecognizer.getSQLType() == SQLType.INSERT) {
-            final boolean oracle = StringUtils.equalsIgnoreCase(JdbcConstants.ORACLE, dbType);
-            final String tableName = sqlRecognizer.getTableName();
-            TableMeta tableMeta;
-            if (oracle) {
-                tableMeta = TableMetaCacheOracle.getTableMeta(getDataSourceProxy(), tableName);
-            } else {
-                tableMeta = TableMetaCache.getTableMeta(getDataSourceProxy(), tableName);
-            }
+            TableMeta tableMeta = TableMetaCacheAdapter.getTableMeta(dbType, getDataSourceProxy(), sqlRecognizer.getTableName(), false);
```
Comment (slievrly) [inline rm-datasource/src/main/java/io/seata/rm/datasource/AbstractConnectionProxy.java]: if sqlRecognizer.getTableName() like "test" and TableMeta contains test without quotes， will it not same tableMeta？

File: rm-datasource/src/main/java/io/seata/rm/datasource/sql/struct/TableMetaAdapter.java [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+        return col;
+    }
+
+    @Override
+    public boolean containsPK(List<String> cols) {
+        if (cols == null) {
+            return false;
+        }
+
+        List<String> pk = getPrimaryKeyOnlyName();
```
Comment (slievrly) [inline rm-datasource/src/main/java/io/seata/rm/datasource/sql/struct/TableMetaAdapter.java]: in this case，it requires pk to contain special symbols, how to guarantee it elsewhere?

File: rm-datasource/src/main/java/io/seata/rm/datasource/exec/InsertExecutor.java [hunk 8]
```diff
@@ -90,8 +89,10 @@ protected TableRecords afterImage(TableRecords beforeImage) throws SQLException
     protected boolean containsPK() {
         SQLInsertRecognizer recognizer = (SQLInsertRecognizer) sqlRecognizer;
         List<String> insertColumns = recognizer.getInsertColumns();
-        TableMeta tmeta = getTableMeta();
-        return tmeta.containsPK(insertColumns);
+        if (insertColumns == null || insertColumns.isEmpty()) {
```
Comment (zjinlei) [inline rm-datasource/src/main/java/io/seata/rm/datasource/exec/InsertExecutor.java]: use seata CollectionUtils

File: codec/seata-codec-seata/src/main/java/io/seata/codec/seata/protocol/transaction/GlobalReportRequestCodec.java [hunk 9]
```diff
@@ -42,7 +42,7 @@
         if (globalStatus != null) {
             out.writeByte((byte)globalStatus.getCode());
         } else {
-            out.writeByte((byte)-1);
+            out.writeByte((byte) -1);
```
Comment (zjinlei) [inline codec/seata-codec-seata/src/main/java/io/seata/codec/seata/protocol/transaction/GlobalReportRequestCodec.java]: restore

File: common/src/test/java/io/seata/common/exception/FrameworkExceptionTest.java [hunk 10]
```diff
@@ -41,7 +41,7 @@ public void testGetErrcode() {
             message.print4();
         });
         assertThat(throwable).hasMessage(FrameworkErrorCode.UnknownAppError.getErrMessage());
-        assertThat(((FrameworkException)throwable).getErrcode()).isEqualTo(FrameworkErrorCode.UnknownAppError);
+        assertThat(((FrameworkException) throwable).getErrcode()).isEqualTo(FrameworkErrorCode.UnknownAppError);
```
Comment (zjinlei) [inline common/src/test/java/io/seata/common/exception/FrameworkExceptionTest.java]: restore this class

File: codec/seata-codec-seata/src/main/java/io/seata/codec/seata/protocol/transaction/GlobalReportRequestCodec.java (hunk 9 above)
Comment (zjinlei) [inline codec/seata-codec-seata/src/main/java/io/seata/codec/seata/protocol/transaction/GlobalReportRequestCodec.java]: restore this class

File: config/seata-config-core/src/main/java/io/seata/config/ConfigurationFactory.java [hunk 11]
```diff
@@ -15,13 +15,13 @@
...
 package io.seata.config;
 
-import java.util.Objects;
-
 import io.seata.common.exception.NotSupportYetException;
 import io.seata.common.loader.EnhancedServiceLoader;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import java.util.Objects;
```
Comment (zjinlei) [inline config/seata-config-core/src/main/java/io/seata/config/ConfigurationFactory.java]: restore this class

File: core/src/main/java/io/seata/core/rpc/netty/NettyPoolableFactory.java [hunk 12]
```diff
@@ -94,12 +94,12 @@ private boolean isResponseSuccess(Object response, NettyPoolKey.TransactionRole
             if (!(response instanceof RegisterTMResponse)) {
                 return false;
             }
-            return ((RegisterTMResponse)response).isIdentified();
+            return ((RegisterTMResponse) response).isIdentified();
         } else if (transactionRole.equals(NettyPoolKey.TransactionRole.RMROLE)) {
             if (!(response instanceof RegisterRMResponse)) {
                 return false;
             }
-            return ((RegisterRMResponse)response).isIdentified();
```
Comment (zjinlei) [inline core/src/main/java/io/seata/core/rpc/netty/NettyPoolableFactory.java]: restore this class

File: core/src/main/java/io/seata/core/rpc/netty/RmMessageListener.java [hunk 13]
```diff
@@ -70,7 +70,7 @@ public void onMessage(RpcMessage request, String serverAddress, ClientMessageSen
         } else if (msg instanceof BranchRollbackRequest) {
             handleBranchRollback(request, serverAddress, (BranchRollbackRequest)msg, sender);
         } else if (msg instanceof UndoLogDeleteRequest) {
-            handleUndoLogDelete((UndoLogDeleteRequest)msg);
+            handleUndoLogDelete((UndoLogDeleteRequest) msg);
```
Comment (zjinlei) [inline core/src/main/java/io/seata/core/rpc/netty/RmMessageListener.java]: restore this class

File: core/src/main/java/io/seata/core/rpc/netty/RpcServer.java [hunk 14]
```diff
@@ -351,7 +351,7 @@ private void handleDisconnect(ChannelHandlerContext ctx) {
     @Override
     public void channelRead(final ChannelHandlerContext ctx, Object msg) throws Exception {
         if (msg instanceof RpcMessage) {
-            RpcMessage rpcMessage = (RpcMessage)msg;
+            RpcMessage rpcMessage = (RpcMessage) msg;
```
Comment (zjinlei) [inline core/src/main/java/io/seata/core/rpc/netty/RpcServer.java]: restore this class

File: core/src/main/java/io/seata/core/store/db/LogStoreDataBaseDAO.java [hunk 15]
```diff
@@ -480,7 +479,7 @@ private ColumnInfo queryTableStructure(final String tableName, String colName) {
         try (Connection conn = logStoreDataSource.getConnection()) {
             DatabaseMetaData dbmd = conn.getMetaData();
             String schema = getSchema(conn);
-            ResultSet tableRs = dbmd.getTables(null, schema, null, new String[] {"TABLE"});
+            ResultSet tableRs = dbmd.getTables(null, schema, null, new String[]{"TABLE"});
```
Comment (zjinlei) [inline core/src/main/java/io/seata/core/store/db/LogStoreDataBaseDAO.java]: restore this class

File: core/src/main/java/io/seata/core/store/db/LogStoreSqls.java [hunk 16]
```diff
@@ -395,8 +395,8 @@ public static String getQureyBranchTransaction(String branchTable, String dbType
     /**
      * Get qurey branch transaction string.
      *
-     * @param branchTable the branch table
-     * @param dbType      the db type
+     * @param branchTable       the branch table
```
Comment (zjinlei) [inline core/src/main/java/io/seata/core/store/db/LogStoreSqls.java]: restore this class

File: discovery/seata-discovery-nacos/src/main/java/io/seata/discovery/registry/nacos/NacosRegistryServiceImpl.java [hunk 17]
```diff
@@ -133,7 +132,7 @@ public void unsubscribe(String cluster, EventListener listener) throws Exception
                         CLUSTER_ADDRESS_MAP.put(clusterName, newAddressList);
                     }
                     subscribe(clusterName, event -> {
```
Comment (zjinlei) [inline discovery/seata-discovery-nacos/src/main/java/io/seata/discovery/registry/nacos/NacosRegistryServiceImpl.java]: restore this class

File: metrics/seata-metrics-core/src/main/java/io/seata/metrics/exporter/ExporterFactory.java [hunk 18]
```diff
@@ -15,17 +15,17 @@
...
 import io.seata.config.ConfigurationFactory;
 import io.seata.core.constants.ConfigurationKeys;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
 /**
```
Comment (zjinlei) [inline metrics/seata-metrics-core/src/main/java/io/seata/metrics/exporter/ExporterFactory.java]: restore this class

File: integration/dubbo-alibaba/src/main/java/io/seata/integration/dubbo/alibaba/TransactionPropagationFilter.java [hunk 19]
```diff
@@ -85,6 +85,7 @@ public Result invoke(Invoker<?> invoker, Invocation invocation) throws RpcExcept
 
     /**
      * get rpc xid
```
Comment (zjinlei) [inline integration/dubbo-alibaba/src/main/java/io/seata/integration/dubbo/alibaba/TransactionPropagationFilter.java]: restore this class

File: integration/dubbo/src/main/java/io/seata/integration/dubbo/TransactionPropagationFilter.java [hunk 20]
```diff
@@ -84,6 +84,7 @@ public Result invoke(Invoker<?> invoker, Invocation invocation) throws RpcExcept
 
     /**
      * get rpc xid
+     *
```
Comment (zjinlei) [inline integration/dubbo/src/main/java/io/seata/integration/dubbo/TransactionPropagationFilter.java]: restore this class

File: integration/grpc/src/main/java/io/seata/integration/grpc/interceptor/server/ServerTransactionInterceptor.java [hunk 21]
```diff
@@ -37,6 +37,7 @@
 
     /**
      * get rpc xid
+     *
```
Comment (zjinlei) [inline integration/grpc/src/main/java/io/seata/integration/grpc/interceptor/server/ServerTransactionInterceptor.java]: restore this class

File: integration/motan/src/main/java/io/seata/integration/motan/MotanTransactionFilter.java [hunk 22]
```diff
@@ -75,6 +75,7 @@ public Response filter(final Caller<?> caller, final Request request) {
 
     /**
      * get rpc xid
+     *
```
Comment (zjinlei) [inline integration/motan/src/main/java/io/seata/integration/motan/MotanTransactionFilter.java]: restore this class

File: integration/sofa-rpc/src/main/java/io/seata/integration/sofa/rpc/TransactionContextConsumerFilter.java [hunk 23]
```diff
@@ -86,6 +86,7 @@ public SofaResponse invoke(FilterInvoker filterInvoker, SofaRequest sofaRequest)
 
     /**
      * get rpc xid
+     *
```
Comment (zjinlei) [inline integration/sofa-rpc/src/main/java/io/seata/integration/sofa/rpc/TransactionContextConsumerFilter.java]: restore this class

File: integration/sofa-rpc/src/main/java/io/seata/integration/sofa/rpc/TransactionContextProviderFilter.java [hunk 24]
```diff
@@ -86,6 +86,7 @@ public SofaResponse invoke(FilterInvoker filterInvoker, SofaRequest sofaRequest)
 
     /**
      * get rpc xid
+     *
```
Comment (zjinlei) [inline integration/sofa-rpc/src/main/java/io/seata/integration/sofa/rpc/TransactionContextProviderFilter.java]: restore this class

File: rm-datasource/src/main/java/io/seata/rm/datasource/undo/AbstractUndoExecutor.java [hunk 25]
```diff
@@ -270,8 +272,8 @@ protected TableRecords queryCurrentRecords(Connection conn) throws SQLException
             replace.append("?,");
         }
         // build check sql
-        String checkSQL = String.format(CHECK_SQL_TEMPLATE, sqlUndoLog.getTableName(), pkName,
-            replace.substring(0, replace.length() - 1));
+        String checkSQL = String.format(CHECK_SQL_TEMPLATE, ColumnUtils.addEscape(sqlUndoLog.getTableName(), getDbType(conn)),
+                tableMeta.getEscapePkName(getDbType(conn)), replace.substring(0, replace.length() - 1));
```
Comment (zjinlei) [inline rm-datasource/src/main/java/io/seata/rm/datasource/undo/AbstractUndoExecutor.java]: extract getDbType(conn)

**Pull request merged**